dio/CHANGELOG.md

```diff
@@ -7,6 +7,8 @@ See the [Migration Guide][] for the complete breaking changes list.**
 
 - Provide fix suggestions for `dart fix`.
 - Fix `receiveTimeout` for streamed responses.
+- Fix cancellation for streamed responses and downloads when using `IOHttpClientAdapter`.
+- Fix receive progress for streamed responses and downloads when using `IOHttpClientAdapter`.
```

Comment:
When does it no longer work? Is it from the recent changes?

Reply:
Cancellation was not implemented. `onReceiveProgress` was only implemented for downloads, not for `ResponseType.Stream`. Our general `onReceiveProgress` handling was done in the transformer which is skipped for streamed responses. This has now been moved to a single location where it is handled for all cases, except for downloads.

The whole stream handling block is also irrelevant for the browser adapter since there is no actual streaming happening there.